streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImplJoin.java

```diff
@@ -155,7 +156,7 @@ public long get() {
 
             final String outerJoinStoreName = userProvidedBaseStoreName == null ? outerJoinStoreGeneratedName : userProvidedBaseStoreName + outerJoinSuffix;
 
-            outerJoinWindowStore = Optional.of(sharedOuterJoinWindowStoreBuilder(outerJoinStoreName, windows, streamJoinedInternal));
+            outerJoinWindowStore = Optional.of(sharedOuterJoinWindowStoreBuilder(outerJoinStoreName, windows, streamJoinedInternal, persistent));
```

Comment:
Just curious: any rationale that we rely on the left store's persistence only to determine the shared store's persistence?

Reply:
No reasons at all. I just chose one side to decide whether go mem or rocks. Left seems the only side that always is True either is Left or Outer. I was thinking of maybe going mem if in left joins, the left side is mem; or if in outer joins, both sides left/right are mem. What do you think?

Reply:
Maybe keeping it simple to always go with left is fine for now, I was just wondering if you have a specific reason for that :)